syntax = "proto3";

package usersprotobuf;

message SavedFile {
    string original_url = 1;
    string original_filename = 2;
    optional string converted_url = 3;
    optional string converted_filename = 4;
}

message UserResponse {
    int32 id = 1;
    string username = 2;
    optional string phone = 3;
    optional string email = 4;
    string first_name = 5;
    string last_name = 6;
    optional string middle_name = 7;
    optional string status = 8;
    bool email_confirmed = 9;
    bool phone_confirmed = 10;
    optional SavedFile avatar = 11;
}

message GetUserByIdRequest {
    int32 id = 1;
}

message GetUserByUsernameRequest {
    string username = 1;
}

message GetUserByEmailRequest {
    string email = 1;
}

message GetUserByTokenRequest {
    string token = 1;
}

message GetUsersByIdsRequest {
    repeated int32 ids = 1;
}

message UsersArrayResponse {
    repeated UserResponse users = 1;
}

service Users {
    rpc GetUserById(GetUserByIdRequest) returns (UserResponse) {}
    rpc GetUsersByIds(GetUsersByIdsRequest) returns (UsersArrayResponse) {}
    rpc GetUserByUsername(GetUserByUsernameRequest) returns (UserResponse) {}
    rpc GetUserByEmail(GetUserByEmailRequest) returns (UserResponse) {}
    rpc GetUserByToken(GetUserByTokenRequest) returns (UserResponse) {}
    rpc GetUserByRefreshToken(GetUserByTokenRequest) returns (UserResponse) {}
}
